LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/LoRaDevice.cs

```diff
@@ -925,18 +929,25 @@ async Task CoreAsync()
             }
         }
 
-        protected virtual void Dispose(bool dispose)
+        internal virtual async Task CloseConnectionAsync(CancellationToken cancellationToken, bool force = false)
         {
-            if (dispose)
+            if ((force || IsConnectionOwner is null or false)
+                && this.connectionManager is { } someConnectionManager)
             {
-                this.connectionManager?.Release(this);
-                this.syncSave.Dispose();
+                await someConnectionManager.GetClient(this).DisconnectAsync(cancellationToken);
             }
         }
 
-        public void Dispose()
+        protected virtual async ValueTask DisposeAsyncCore()
+        {
+            if (this.connectionManager is { } someConnectionManager)
+                await someConnectionManager.ReleaseAsync(this);
+            this.syncSave.Dispose();
+        }
+
+        public async ValueTask DisposeAsync()
```

Comment:
Unless I missed it, there doesn't seem to be a test that checks the release on connection manager when the device is disposed.

Reply:
Good catch. Fixed with 19db0350a45342831cfdc7189135d59f02ef96b1
Resolve if ok